### PR TITLE
[Flink] Optimising the calculation logic for average throughput and updating nexmark  query

### PR DESCRIPTION
**Pull request by lsyldliu**

### CODE REVIEW

File: nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java [hunk 1]
```diff
@@ -154,29 +168,30 @@ public JobBenchmarkMetric reportMetric(String jobId, long eventsNum) {
...
-					metric.getPrettyTps(),
-					metric.getPrettyCpu());
-		} else {
-			message = String.format("Summary Average: EventsNum=%s, Cores=%s, Time=%s s",
-					NUMBER_FORMAT.format(eventsNum),
-					metric.getPrettyCpu(),
-					formatDoubleValue(metric.getTimeSeconds()));
-		}
-
+		message = String.format("Summary Average: Throughput=%s, Cores=%s",
```
Comment (masteryhx) [inline nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java:187]: We could aggregate two messages to one.

File: nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java [hunk 2]
```diff
@@ -154,29 +168,30 @@ public JobBenchmarkMetric reportMetric(String jobId, long eventsNum) {
 		}
 		double sumTps = 0.0;
 		double sumCpu = 0.0;
+		int realMetricSize = metrics.size();
 
 		for (BenchmarkMetric metric : metrics) {
+			if (Double.compare(metric.getTps(), 0.0) == 0) {
```
Comment (masteryhx) [inline nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java:174]: Only backmost zero should be removed.

File: nexmark-flink/src/main/java/com/github/nexmark/flink/workload/Workload.java [hunk 3]
```diff
@@ -103,21 +103,12 @@ public long getWarmupEvents() {
 
 	public void validateWorkload(Duration monitorDuration) {
 		boolean unboundedMonitor = monitorDuration.toMillis() == Long.MAX_VALUE;
-		if (getEventsNum() == 0) {
-			// TPS mode
```
Comment (masteryhx) [inline nexmark-flink/src/main/java/com/github/nexmark/flink/workload/Workload.java:107]: Some parameters should also be removed? 
e.g. nexmark.metric.monitor.duration

**Review by masteryhx — CHANGES_REQUESTED** [review]
Thanks for the pr!
I have lefted some comments, PTAL.

File: nexmark-flink/src/main/java/com/github/nexmark/flink/workload/Workload.java (hunk 3 above)
Comment (lsyldliu) [reply nexmark-flink/src/main/java/com/github/nexmark/flink/workload/Workload.java:107]: It also has been used in test, so I think we don't need to remove it now.

File: nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java (hunk 2 above)
Comment (lsyldliu) [reply nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java:174]: I don't think this has much impact on the TPS calculation, we can ignore this optimisation for now.

**Comment by lsyldliu** [comment]
@masteryhx Thanks for your review, I have updated.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request closed without merging**